code/go/internal/validator/semantic/validate_integration_policy_template_path.go

```diff
@@ -148,18 +150,19 @@ func validateInputWithStreams(fsys fspath.FS, input string, dsMap map[string]dat
 				stream.TemplatePath = defaultStreamTemplatePath
 			}
 
-			_, err := fs.ReadFile(fsys, path.Join(dsDir, "agent", "stream", stream.TemplatePath))
+			dir := path.Join(dsDir, "agent", "stream")
+			foundFile, err := findPathWithPattern(fsys, dir, stream.TemplatePath)
+			if err != nil {
+				return err
+			}
+			if foundFile == "" {
+				return errTemplateNotFound
+			}
+
+			_, err = fs.ReadFile(fsys, foundFile)
 			if err != nil {
 				if errors.Is(err, fs.ErrNotExist) {
```

Comment:
Nit. This should not happen, right? Though we can leave this in case the file is somehow removed between finding and opening it.

Reply:
this was supposed to be removed, my bad i did not remove it when changing the code.